### PR TITLE
[rolling] image_publisher: Fix loading of the camera info parameters on startup

### PR DESCRIPTION
**Pull request by Kotochleb**
As described in https://github.com/ros-perception/image_pipeline/issues/965 camera info is not loaded from the file on node initialization, but only when the parameter is reloaded.

This PR resolves this issue and should be straightforward to port it to `Humble`, `Iron` and `Jazzy`.

### CODE REVIEW

**Review by ahcorde — COMMENTED** [review]
can you target `rolling` ? then I will backport this to the other distros

**Comment by Kotochleb** [comment]
@ahcorde the changes were retargeted to `rolling` as requested.

File: image_publisher/src/image_publisher.cpp [hunk 1]
```diff
@@ -71,36 +71,46 @@ ImagePublisher::ImagePublisher(
   auto param_change_callback =
     [this](std::vector<rclcpp::Parameter> parameters) -> rcl_interfaces::msg::SetParametersResult
     {
+      RCLCPP_INFO(get_logger(), "param_change_callback");
```
Comment (ahcorde) [inline image_publisher/src/image_publisher.cpp]: remove

File: image_publisher/src/image_publisher.cpp [hunk 2]
```diff
@@ -224,9 +234,7 @@ void ImagePublisher::onInit()
   camera_info_.p = {1, 0, static_cast<float>(camera_info_.width / 2), 0, 0, 1,
     static_cast<float>(camera_info_.height / 2), 0, 0, 0, 1, 0};
 
-  timer_ = this->create_wall_timer(
```
Comment (ahcorde) [inline image_publisher/src/image_publisher.cpp:244]: if you remove these lines the `dowork` method will never be called and the image will be never flipped

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: image_publisher/src/image_publisher.cpp (hunk 1 above)
Comment (Kotochleb) [reply image_publisher/src/image_publisher.cpp]: Done

File: image_publisher/src/image_publisher.cpp (hunk 2 above)
Comment (Kotochleb) [reply image_publisher/src/image_publisher.cpp:244]: The timer is set in the `reconfigureCallback`. I didn't want to repeat the `CameraInfoManager` block of code in `onInit` and did not want to create timer there just to recreate it once again in `reconfigureCallback`.

**Comment by Kotochleb** [comment]
@ahcorde FYI changes from https://github.com/ros-perception/image_pipeline/pull/985 are already merged in here.

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport jazzy humble iron

**Pull request merged**